# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1529940701 14400
#      Mon Jun 25 11:31:41 2018 -0400
# Node ID 3b5db971e091772a11e943a3d7baee7f87f74432
# Parent  7cdda3ad947fa17bd5e198059a25c61952203100
ensure test/mex directory exists when compiling mex test files

* test/mex/module.mk (%.mex : %.c): Limit pattern rule to
$(MEX_TEST_FUNCTIONS) files.  Depend on %reldir%/$(octave_dirstamp) as
an order-only prerequisite.
(DIRSTAMP_FILES): Add %reldir%/$(octave_dirstamp) to the list.

diff --git a/test/mex/module.mk b/test/mex/module.mk
--- a/test/mex/module.mk
+++ b/test/mex/module.mk
@@ -34,14 +34,16 @@ MKOCTFILECPPFLAGS = \
 MKOCTFILE = $(top_builddir)/src/mkoctfile $(MKOCTFILECPPFLAGS)
 
 MKMEXFILECPPFLAGS = \
   -I$(top_srcdir)/libinterp/corefcn \
   -Ilibinterp/corefcn
 
 MKMEXFILE = $(top_builddir)/src/mkoctfile --mex $(MKMEXFILECPPFLAGS)
 
-%.mex : %.c
+$(MEX_TEST_FUNCTIONS) : %.mex : %.c | %reldir%/$(octave_dirstamp)
 	$(AM_V_mkmexfile)$(MKMEXFILE) $(AM_VOPT_mkmexfile) $< -o $@ || rm -f $@
 
+DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
+
 ## Until we decide how to handle installing the executable MEX files,
 ## don't install them or the associated test files.
 noinst_TEST_FILES += $(mex_TEST_FILES)
